boilerplate/app/models/EpisodeStore.ts

```diff
@@ -18,6 +19,24 @@ export const EpisodeStoreModel = types
         console.tron.error(`Error fetching episodes: ${JSON.stringify(response)}`, [])
       }
     },
+    addFavorite(episode: Episode) {
+      store.favorites.set(episode.guid, clone(episode))
+    },
+    removeFavorite(episode: Episode) {
+      store.favorites.delete(episode.guid)
+    },
+    hasFavorite(episode: Episode) {
+      return store.favorites.has(episode.guid)
+    },
+  }))
+  .actions((store) => ({
+    toggleFavorite(episode: Episode) {
+      if (store.favorites.has(episode.guid)) {
+        store.removeFavorite(episode)
```

Comment:
Changing to an array of references means this part would change somewhat. Probably `.find(ep => ep.guid === episode.guid)`

Reply:
`includes` should work